agent/doc/agent-doc.md

```diff
@@ -0,0 +1,23 @@
+# Centreon Agent documentation {#mainpage}
+
+## Introduction
+
+The purpose of this program is to run checks in Windows and Linux operating systems. It is entirely asynchronous, with the exception of the gRPC layers. It is also single-threaded and therefore needs no mutexes, except in the gRPC part.
+This is why, when a request is received, it is posted to ASIO for processing in the main thread.
+
+## Configuration
+configuration is given by Engine by a AgentConfiguration sent over grpc
+The configuration object is embedded in MessageToAgent::config
+
+## Scheduler
+We trie to spread checks over check_period.
+Example: We have 10 checks to execute during one second. check1 will start at now, second at now + 0.1s..
+
+When Agent receives configuration, all checks are recreated.
+For example, we have 100 checks to execute in 10 minute, at it is 12:00:00.
+First  service check will start right now, second one at 12:00:06, third at 12:00:12... and the last one at 12:09:54
+We don't care about tests duration, we work with time points. 
+In the previous example, time of second check of first service will be scheduled at 12:00:10 even if all other checks has not been yet started.
+
+In case of check duration is too long, we might exceed maximum of concurrent checks. In that case checks will b executed as soon one will be ended.
+So second check may start later than scheduled time point (12:00:10) if other first checks are too long. Order of checks is always respected even in case of bottleneck.
```

Comment:
```suggestion
This means that the second check may start later than the scheduled time point (12:00:10) if the other first checks are too long. The order of checks is always respected even in case of a bottleneck.
```